src/implicit_function.jl

```diff
@@ -1,3 +1,21 @@
+struct Forward{returns_byproduct,F}
+    f::F
+    function Forward{returns_byproduct}(f::F) where {returns_byproduct,F}
+        return new{returns_byproduct,F}(f)
+    end
+end
+(f::Forward{true})(x; kwargs...) = f.f(x; kwargs...)
+(f::Forward{false})(x; kwargs...) = (f.f(x; kwargs...), 0)
+
+struct Conditions{accepts_byproduct,F}
```

Comment:
Maybe instead of `return/accept_byproduct` we could just say `byproduct`, or `BP` for short?